drivers/gpio/gpio_stm32.c

```diff
@@ -41,24 +45,37 @@ static void gpio_stm32_isr(int line, void *arg)
  */
 const int gpio_stm32_flags_to_conf(int flags, int *pincfg)
 {
-	int direction = flags & GPIO_DIR_MASK;
-	int pud = flags & GPIO_PUD_MASK;
 
 	if (pincfg == NULL) {
 		return -EINVAL;
 	}
 
-	if (direction == GPIO_DIR_OUT) {
+	if ((flags & GPIO_OUTPUT) != 0) {
+
 		*pincfg = STM32_PINCFG_MODE_OUTPUT;
+
+		if ((flags & GPIO_PUSH_PULL) != 0) {
```

Comment:
This check is wrong: `GPIO_PUSH_PULL` has value zero, the branch will never be taken.

Review `include/dt-bindings/gpio/gpio.h`.

Reply:
Thanks!